### PR TITLE
fix: Validate base URL in auth fragments

### LINKED ISSUE #1490: Authentication fragments do not validate base URL
**Describe the bug**
Authentication fragments (Login, Signup, Start and Reset Password) do not validate base URL.

**To Reproduce**
1. Go to Login.
2. Enter wrong input in the Other URL field.
3. It will be accepted.

**Expected behavior**
Error should have been displayed.

**Would you like to work on the issue?**
- [x] Yes
- [ ] No

### PR DESCRIPTION
**Pull request by ShridharGoel**
Fixes #1490 

Checklist:

- [ ] I have checked for PMD and check-style issues 
- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream development branch.

Changes: 

Now, base URL entered by the user would be validated and error would be shown if it's found to be invalid.

GIF for the change:

ezgif com-video-to-gif 33

### CODE REVIEW

File: app/src/main/res/layout/auth_url_container.xml [hunk 1]
```diff
@@ -44,6 +44,7 @@
                 android:hint="@string/other_url"
                 android:inputType="textUri"
                 android:imeOptions="actionGo"
+                app:validateType='@{"url"}'
```
Comment (Masquerade0097) [inline app/src/main/res/layout/auth_url_container.xml]: Check if adding `validateType` makes adding this URL mandatory.

Comment (ShridharGoel) [reply app/src/main/res/layout/auth_url_container.xml]: Yes, that's the problem currently. Hence, I had added WIP in the title after I realized this.

Comment (ShridharGoel) [reply app/src/main/res/layout/auth_url_container.xml]: Please review now.

**Comment by iamareebjamal** [comment]
Too complex change

**Comment by ShridharGoel** [comment]
@iamareebjamal I think it's looking complex because the change had to be implemented in all the fragments which have the functionality of changing the base URL. I think there's no simpler way to do it, is there one? Validating in XML wouldn't work since it validates the view even when it's hidden.

**Comment by ShridharGoel** [comment]
@iamareebjamal Please review this

**Pull request merged**